src/main/java/de/tum/in/www1/artemis/domain/Result.java

```diff
@@ -210,7 +212,7 @@ public void setRated(Boolean rated) {
         this.rated = rated;
     }
 
-    public void setRatedIfNotExceeded(@Nullable ZonedDateTime exerciseDueDate, ZonedDateTime submissionDate) {
+    public void setRatedIfNotAfterDueDate(@Nullable ZonedDateTime exerciseDueDate, @NotNull ZonedDateTime submissionDate) {
         this.rated = exerciseDueDate == null || submissionDate.isBefore(exerciseDueDate) || submissionDate.isEqual(exerciseDueDate);
```

Comment:
Why is no grace period added here?

Reply:
Because we decided that we only want to apply that to programming exercises (where a delay is possible due to the usage of external services and the need to change access rights). Other exercises are directly submitted on Artemis and don't have these issues.